app/models/test_track/identity_session_discriminator.rb

```diff
@@ -13,6 +13,10 @@ def participate_in_online_session?
     authenticated_resource_matches_identity?
   end
 
+  def participate_in_unauthenticated_session?
+    web_context? && !controller.respond_to?(authenticated_resource_method_name, true)
```

Comment:
Maybe if we made the else side raise as I'm suggesting above you'd want to raise when trying to call sign_up from an authenticated controller, but I could imagine situations where auth frameworks actually do define and set that value on the controller. We should maybe stay out of it and just do our best to sign up if we're in a web_context?